ably/error.go

```diff
@@ -144,6 +145,31 @@ func errFromUnprocessableBody(resp *http.Response) error {
 	return &ErrorInfo{Code: ErrBadRequest, StatusCode: resp.StatusCode, err: err}
 }
 
+func isTimeoutOrDnsErr(err error) bool {
+	if err == nil {
+		return false
+	}
+	switch err := err.(type) {
```

Comment:
It's almost always wrong to unwrap errors this way. Errors are supposed to follow a "chain" convention. See https://pkg.go.dev/errors

This should also the special cases here unnecessary, because `*net.DNSError`, `*url.Error` and the timeout errors returned by the net/http package can all be unwrapped to a `net.Error`.

Reply:
Referring to https://blog.golang.org/go1.13-errors. Seems new way of handling errors have been introduced, since we support minimum 1.13, I will use new ways of handling errors